babel/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -199,6 +199,143 @@ SqlColumnAttribute ColumnAttributeDefault() :
     }
 }
 
+/**
+ * Parses a column attribute specified by the GENERATED statement.
+ */
+SqlColumnAttribute ColumnAttributeGenerated() :
+{
+    final GeneratedType generatedType;
+    final List<SqlColumnAttributeGeneratedOption> generateOptions =
+        new ArrayList<SqlColumnAttributeGeneratedOption>();
+}
+{
+    <GENERATED>
+    (
+        <ALWAYS> { generatedType = GeneratedType.ALWAYS; }
+    |
+        <BY> <DEFAULT_> { generatedType = GeneratedType.BY_DEFAULT; }
+    )
+    <AS> <IDENTITY>
+    [
+        <LPAREN>
+        (
+            ColumnAttributeGeneratedOption(generateOptions)
+        )+
+        <RPAREN>
+    ]
+    {
+        return new SqlColumnAttributeGenerated(getPos(), generatedType,
+            generateOptions);
+    }
+}
+
+/**
+ * Parses an option specified for the GENERATED column attribute, and adds it
+ * to a list.
+ */
+void ColumnAttributeGeneratedOption(
+    List<SqlColumnAttributeGeneratedOption> generateOptions) :
```

Comment:
nit: generatedOptions to be consistent with the function name and class type? (same for local variable)